Source/WebCore/Modules/WebGPU/GPUQueue.h

```diff
@@ -60,7 +60,7 @@ class GPUQueue : public RefCounted<GPUQueue> {
     using OnSubmittedWorkDonePromise = DOMPromiseDeferred<IDLNull>;
     void onSubmittedWorkDone(OnSubmittedWorkDonePromise&&);
 
-    void writeBuffer(
+    ExceptionOr<void> writeBuffer(
```

Comment:
Is this correct? It throws, instead of using GPUErrorScope?

Reply:
Yes that is correct:
```
If any of the following conditions are unsatisfied, throw [OperationError](https://webidl.spec.whatwg.org/#operationerror) and stop.

contentsSize ≥ 0.
dataOffset + contentsSize ≤ dataSize.
contentsSize, converted to bytes, is a multiple of 4 bytes.
```

The specification is quite confusing in this regard. Some things throw exceptions, other items use GPUErrorScope. I'm trying to understand why there are these two methods of error handling but perhaps a specification issue is warranted.